File: SunburstHierarchicalDataAdapterSample/SunburstHierarchicalDataAdapterSample/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraTreeMap;

namespace SunburstHierarchicalDataAdapterSample {
    public partial class MainForm : XtraForm {
        public MainForm() {
            InitializeComponent();
        }

        private void OnFormLoad(object sender, EventArgs e) {
            SunburstHierarchicalDataAdapter dataAdapter = new SunburstHierarchicalDataAdapter();
            dataAdapter.Mappings.Add(new SunburstHierarchicalDataMapping {
                Type = typeof(CountryInfo),
                LabelDataMember = "CountryName",
                ChildrenDataMember = "CityInfos"
            });
            dataAdapter.Mappings.Add(new SunburstHierarchicalDataMapping {
                Type = typeof(CityInfo),
                LabelDataMember = "CityName",                
                ChildrenDataMember = "SaleInfos"
            });
            dataAdapter.Mappings.Add(new SunburstHierarchicalDataMapping {
                Type = typeof(ProductInfo),
                ValueDataMember = "Total",
                LabelDataMember = "Category",
            });
            dataAdapter.DataSource = CreateInfos();
            sunburstControl.DataAdapter = dataAdapter;

            sunburstControl.Colorizer = new SunburstPaletteColorizer {
                ColorizeGroups = true,
                Palette = Palette.Office2019Palette
            };

            sunburstControl.CenterLabel.TextPattern = "Total:" + Environment.NewLine + "{TV}";
          
        }
        List<CountryInfo> CreateInfos() {
            List<CountryInfo> infos = new List<CountryInfo>();

            CountryInfo germanyInfo = new CountryInfo { CountryName = "Germany" };
            CityInfo leipzigInfo = new CityInfo { CityName = "Leipzig" };
            leipzigInfo.SaleInfos.AddRange(new List<ProductInfo> {
                new ProductInfo { Category = "Beverages", Total = 2634.5 },
                new ProductInfo { Category = "Baked Goods", Total = 4523.98 },
                new ProductInfo { Category = "Grains and Cereals", Total = 913.85 },
                new ProductInfo { Category = "Milk and Dairy", Total = 4219.98 },
            });
            CityInfo berlinInfo = new CityInfo { CityName = "Berlin" };
            berlinInfo.SaleInfos.AddRange(new List<ProductInfo> {
                new ProductInfo { Category = "Frozen Foods", Total = 900 }
            });
            germanyInfo.CityInfos.AddRange( new List<CityInfo> { leipzigInfo, berlinInfo } );

            CountryInfo spainInfo = new CountryInfo { CountryName = "Spain" };
            CityInfo barcelonaInfo = new CityInfo { CityName = "Barcelona" };
            barcelonaInfo.SaleInfos.AddRange(new List<ProductInfo> {
                new ProductInfo { Category = "Baked Goods", Total = 1239.2 },
                new ProductInfo { Category = "Fruits", Total = 450.41 },
                new ProductInfo { Category = "Milk and Dairy", Total = 692.5 },
            });
            CityInfo madridInfo = new CityInfo { CityName = "Madrid" };
            madridInfo.SaleInfos.AddRange(new List<ProductInfo> {
                new ProductInfo { Category = "Spices", Total = 1010.30 },
                new ProductInfo { Category = "Vegetables", Total = 2078 }
            });
            spainInfo.CityInfos.AddRange(new List<CityInfo> { barcelonaInfo, madridInfo });

            infos.AddRange(new List<CountryInfo> { germanyInfo, spainInfo });

            return infos;
        }
    }
}

File: VB/SunburstHierarchicalDataAdapterSample/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraTreeMap

Namespace SunburstHierarchicalDataAdapterSample
    Partial Public Class MainForm
        Inherits XtraForm

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub OnFormLoad(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
            Dim dataAdapter As New SunburstHierarchicalDataAdapter()
            dataAdapter.Mappings.Add(New SunburstHierarchicalDataMapping With { _
                .Type = GetType(CountryInfo), _
                .LabelDataMember = "CountryName", _
                .ChildrenDataMember = "CityInfos" _
            })
            dataAdapter.Mappings.Add(New SunburstHierarchicalDataMapping With { _
                .Type = GetType(CityInfo), _
                .LabelDataMember = "CityName", _
                .ChildrenDataMember = "SaleInfos" _
            })
            dataAdapter.Mappings.Add(New SunburstHierarchicalDataMapping With { _
                .Type = GetType(ProductInfo), _
                .ValueDataMember = "Total", _
                .LabelDataMember = "Category" _
            })
            dataAdapter.DataSource = CreateInfos()
            sunburstControl.DataAdapter = dataAdapter

            sunburstControl.Colorizer = New SunburstPaletteColorizer With {.Palette = Palette.Office2019Palette}

            sunburstControl.CenterLabel.TextPattern = "Total:" & Environment.NewLine & "{TV}"

        End Sub
        Private Function CreateInfos() As List(Of CountryInfo)
            Dim infos As New List(Of CountryInfo)()

            Dim germanyInfo As CountryInfo = New CountryInfo With {.CountryName = "Germany"}
            Dim leipzigInfo As CityInfo = New CityInfo With {.CityName = "Leipzig"}
            leipzigInfo.SaleInfos.AddRange(New List(Of ProductInfo) From { _
                New ProductInfo With { _
                    .Category = "Beverages", _
                    .Total = 2634.5 _
                }, _
                New ProductInfo With { _
                    .Category = "Baked Goods", _
                    .Total = 4523.98 _
                }, _
                New ProductInfo With { _
                    .Category = "Grains and Cereals", _
                    .Total = 913.85 _
                }, _
                New ProductInfo With { _
                    .Category = "Milk and Dairy", _
                    .Total = 4219.98 _
                } _
            })
            Dim berlinInfo As CityInfo = New CityInfo With {.CityName = "Berlin"}
            berlinInfo.SaleInfos.AddRange(New List(Of ProductInfo) From { _
                New ProductInfo With { _
                    .Category = "Frozen Foods", _
                    .Total = 900 _
                } _
            })
            germanyInfo.CityInfos.AddRange(New List(Of CityInfo) From {leipzigInfo, berlinInfo})

            Dim spainInfo As CountryInfo = New CountryInfo With {.CountryName = "Spain"}
            Dim barcelonaInfo As CityInfo = New CityInfo With {.CityName = "Barcelona"}
            barcelonaInfo.SaleInfos.AddRange(New List(Of ProductInfo) From { _
                New ProductInfo With { _
                    .Category = "Baked Goods", _
                    .Total = 1239.2 _
                }, _
                New ProductInfo With { _
                    .Category = "Fruits", _
                    .Total = 450.41 _
                }, _
                New ProductInfo With { _
                    .Category = "Milk and Dairy", _
                    .Total = 692.5 _
                } _
            })
            Dim madridInfo As CityInfo = New CityInfo With {.CityName = "Madrid"}
            madridInfo.SaleInfos.AddRange(New List(Of ProductInfo) From { _
                New ProductInfo With { _
                    .Category = "Spices", _
                    .Total = 1010.30 _
                }, _
                New ProductInfo With { _
                    .Category = "Vegetables", _
                    .Total = 2078 _
                } _
            })
            spainInfo.CityInfos.AddRange(New List(Of CityInfo) From {barcelonaInfo, madridInfo})

            infos.AddRange(New List(Of CountryInfo) From {germanyInfo, spainInfo})

            Return infos
        End Function
    End Class
End Namespace
